server/api/src/rsDataObjRsync.cpp

```diff
@@ -261,7 +267,41 @@ rsRsyncDataToData( rsComm_t *rsComm, dataObjInp_t *dataObjInp ) {
         /* use rsDataObjChksum because the path could be in a remote zone */
         status = rsDataObjChksum(rsComm, &dataObjCopyInp.destDataObjInp, &destChksumStr);
 
-        if (status < 0 && status != CAT_NO_ACCESS_PERMISSION && status != CAT_NO_ROWS_FOUND) {
+        if (CHECK_VERIFICATION_RESULTS == status) {
+            try {
+                const auto& verification_result = json::parse(destChksumStr);
+                for (const auto& entry : verification_result) {
+                    if (entry.at("error_code") != 0) {
+                        // "actual" error happened: log it and fail
+                        log_api::error("{}: error in rsDataObjChksum, message = [{}]; error_code = [{}]",
+                                       __func__,
+                                       entry.at("message").get<std::string>(),
+                                       entry.at("error_code").get<int>());
+                        // NOLINTNEXTLINE(cppcoreguidelines-no-malloc, cppcoreguidelines-owning-memory)
+                        free(srcChksumStr);
+                        // NOLINTNEXTLINE(cppcoreguidelines-no-malloc, cppcoreguidelines-owning-memory)
+                        free(destChksumStr);
+                        clearKeyVal(&dataObjCopyInp.srcDataObjInp.condInput);
+                        clearKeyVal(&dataObjCopyInp.destDataObjInp.condInput);
+                        return status;
+                    }
+                }
+            }
+            catch (const json::exception& ex) {
+                log_api::error("{}: json exception from rsDataObjChksum return, destChksumStr = \"{}\"; what = [{}]",
+                               __func__,
+                               destChksumStr,
```

Comment:
I don't think nullptr is understood by the logger or fmtlib. Let's verify and update this code accordingly.